dotnet-efcore/RecordsAsModelClasses/RecordsAsModelClasses/Controllers/v1/CarsController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using RecordsAsModelClasses.Core.Context;
+using RecordsAsModelClasses.Core.DTOs;
+using RecordsAsModelClasses.Core.Entities.Records;
+
+namespace RecordsAsModelClasses.Controllers.v1;
+
+[Route("api/v1/cars")]
+[ApiController]
+public class CarsController : ControllerBase
+{
+    private readonly CarDbContext _context;
+
+    public CarsController(CarDbContext context)
+    {
+        _context = context;
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> CreateCarAsync([FromBody] CarDto carDto)
+    {
+        var car = new Car(carDto.Id, carDto.Make, carDto.Model, carDto.Year);
+        
+        _context.RecordCars.Add(car);
+        await _context.SaveChangesAsync();
+        
+        return Created("api/v1/cars", car);
+    }
+
+    [HttpPut("{id:int}")]
+    public async Task<IActionResult> UpdateCar(int id, [FromBody] Car updatedCar)
+    {
+        var car = await _context
+            .RecordCars
+            .Where(c => c.Id == id)
+            .FirstOrDefaultAsync();
+
+        if (car == null)
+        {
+            return NotFound();
+        }
+
+        _context.Entry(car).CurrentValues.SetValues(updatedCar);
+
+        await _context.SaveChangesAsync();
+        return NoContent();
+    }
+
+    [HttpGet("{id:int}")]
+    public async Task<IActionResult> GetCar(int id)
+    {
+        var car = await _context.RecordCars.FindAsync(id);
+
+        if (car == null)
+        {
+            return NotFound();
+        }
+
+        var carDto = new CarDto(car.Id, car.Make, car.Model, car.Year);
+
+        return CreatedAtAction(nameof(GetCar), new { id}, carDto);
```

Comment:
This should return Ok(carDto); the `CreatedAtAction` should be returned by creation endpoints : 
```suggestion
        return Ok(carDto);
```